packages/advanced-logic/src/extensions/payment-network/address-based.ts

```diff
@@ -1,292 +1,279 @@
+  ): ExtensionTypes.IAction {
+    if (
+      addPaymentAddressParameters.paymentAddress &&
+      !this.isValidAddress(addPaymentAddressParameters.paymentAddress)
+    ) {
+      throw Error('paymentAddress is not a valid address');
+    }
 
-/**
- * Applies the extension action to the request
- * Is called to interpret the extensions data when applying the transaction
- *
- * @param isValidAddress address validator function
- * @param extensionsState previous state of the extensions
- * @param extensionAction action to apply
- * @param requestState request state read-only
- * @param actionSigner identity of the signer
- *
- * @returns state of the request updated
- */
-function applyActionToExtension(
-  isValidAddress: (address: string) => boolean,
-  extensionsState: RequestLogicTypes.IExtensionStates,
-  extensionAction: ExtensionTypes.IAction,
-  requestState: RequestLogicTypes.IRequest,
-  actionSigner: IdentityTypes.IIdentity,
-  timestamp: number,
-): RequestLogicTypes.IExtensionStates {
-  const copiedExtensionState: RequestLogicTypes.IExtensionStates = Utils.deepCopy(extensionsState);
+    return {
+      action: ExtensionTypes.PnAddressBased.ACTION.ADD_PAYMENT_ADDRESS,
+      id: this.extensionId,
+      parameters: {
+        paymentAddress: addPaymentAddressParameters.paymentAddress,
+      },
+    };
+  }
 
-  if (extensionAction.action === ExtensionTypes.PnAddressBased.ACTION.CREATE) {
-    if (requestState.extensions[extensionAction.id]) {
-      throw Error(`This extension has already been created`);
+  /**
+   * Creates the extensionsData to add a refund address
+   *
+   * @param extensions extensions parameters to create
+   *
+   * @returns IAction the extensionsData to be stored in the request
+   */
+  public createAddRefundAddressAction(
+    addRefundAddressParameters: ExtensionTypes.PnAddressBased.IAddRefundAddressParameters,
+  ): ExtensionTypes.IAction {
+    if (
+      addRefundAddressParameters.refundAddress &&
+      !this.isValidAddress(addRefundAddressParameters.refundAddress)
+    ) {
+      throw Error('refundAddress is not a valid address');
     }
 
-    copiedExtensionState[extensionAction.id] = applyCreation(
-      isValidAddress,
-      extensionAction,
-      timestamp,
-    );
-
-    return copiedExtensionState;
+    return {
+      action: ExtensionTypes.PnAddressBased.ACTION.ADD_REFUND_ADDRESS,
+      id: this.extensionId,
+      parameters: {
+        refundAddress: addRefundAddressParameters.refundAddress,
+      },
+    };
   }
 
-  // if the action is not "create", the state must have been created before
-  if (!requestState.extensions[extensionAction.id]) {
-    throw Error(`The extension should be created before receiving any other action`);
-  }
+  /**
+   * Applies a creation
+   *
+   * @param isValidAddress address validator function
+   * @param extensionAction action to apply
+   *
+   * @returns state of the extension created
+   */
```

Comment:
```suggestion
```
The params don't match and I don't think comments are useful for the implementation of an abstract class method (+ it's protected).